src/component/framework/conditional.js

```diff
@@ -27,3 +45,7 @@ export class Conditional {
   }
 
 }
+
+function isPromise(value) {
+  return (typeof value === 'object' && value.then);
```

Comment:
Feels like that would require people to use Promise constructor of the global. I would want people to decide for themselves what promise lib they would like to use.